test/javascripts/turbohead_test.coffee

```diff
@@ -120,3 +133,22 @@ describe 'TurboHead', ->
         newRequest(['a.js', 'b.js']).then ->
           assertScriptCount(2)
           assert.isTrue(requests[1].isFulfilled)
+
+  describe 'script errors', ->
+    it 'sends events for script errors', ->
+      promise = new Promise (resolve) ->
+        $(document).one('page:script-error', (evt) ->
+          assert.equal(evt.originalEvent.data.url, 'a.js')
+          resolve()
+        )
+
+      newRequest(['a.js'])
+      failScriptDownload()
+      promise
+
+    it 'fulfills requests with script errors', ->
+      newRequest(['a.js', 'b.js', 'c.js'])
```

Comment:
I know this is the way this entire file is done, but figured I'd still comment for posterity: the usage of a closure variables here feels really, really awkward. I feel like these would be so much easier to read if they returned things explicitly, rather than appending/ reading from a queue/ adding promises to a queue. Something along the lines of (just a rough outline):

```js
request = createTestRequest(scripts: ['a.js', 'b.js', 'c.js'])
Promise.all(
  request.finishScriptDownload('a.js'),
  request.failScriptDownload('b.js'),
  request.finishScriptDownload('c.js')
)
  .then(() -> assert.isTrue(request.isFulfilled))
```